app/controllers/webhook_controller.rb

```diff
@@ -23,19 +24,33 @@ def callback
       case event
       when Line::Bot::Event::Message
         case event.type
-        when Line::Bot::Event::MessageType::Text
-          message = {
-            type: 'text',
-            text: event.message['text']
-          }
-          client.reply_message(event['replyToken'], message)
-        when Line::Bot::Event::MessageType::Image, Line::Bot::Event::MessageType::Video
-          response = client.get_message_content(event.message['id'])
-          tf = Tempfile.open("content")
-          tf.write(response.body)
+        when Line::Bot::Event::MessageType::Image
+          image_content = client.get_message_content(event.message['id']).body
+          text = get_text_in_image(image_content)
+        else
+          text = '画像を送信してください。'
         end
+        message = { type: 'text', text: text }
+        client.reply_message(event['replyToken'], message)
       end
     }
     head :ok
   end
+
+  private
+  def get_text_in_image(image_content)
+    image_annotator = Google::Cloud::Vision::ImageAnnotator.new
+    requests = [{
+      image: { content: image_content },
+      features: [{ type: 'DOCUMENT_TEXT_DETECTION' }]
+    }]
+    response = image_annotator.batch_annotate_images(requests)
+    text_annotations = response.responses[0].text_annotations
+    if text_annotations.empty?
+      text = '文字が検出されませんでした。'
+    else
+      text = text_annotations[0].description.chomp
+    end
+    text
```

Comment:
[nits]
ruby の場合、最後に評価したバリューが返り値になるので、最後の一行がなくても `text` が返ってくれる、はず?
後、else外してこんな感じで書いてもスマートでいいかもしれないですね（感想）。
```suggestion
    if text_annotations.empty?
      '文字が検出されませんでした。'
    end
    
    text_annotations[0].description.chomp
```

あと、好み別れるんですが、後置if が使えるので、このぐらいコード量減らせたりもします。
(これは分岐が増えると辛いのです)
```suggestion
    return '文字が検出されませんでした。'    if text_annotations.empty?
    text_annotations[0].description.chomp
```

Reply:
僕これ完全に同じ指摘しようとしてましたね・・・。ハッピーアイスクリーム。
そも論 `Line::Bot::Event::MessageType::Image`で分岐してからきてるので、今後分岐が増える可能性は少ないため一番下のケースを個人的に推します

Reply:
ありgとうございます。
後置if使ってみます！

Reply:
[imo] 自分ならこう！

```ruby
text_annotations[0]&.description&.chomp || '文字が検出されませんでした。'
```

Reply:
イケメン

Reply:
&. というものがあるんですね。
知らなかったです。
ありがとうございます！

Reply:
やるわ
(なるほど、空の配列にアクセスしてもnilが返ってくるだけで、範囲外アクセスで落ちたりしないのか...)